dotCMS/src/main/java/com/dotcms/tika/TikaUtils.java

```diff
@@ -138,6 +148,156 @@ public boolean generateMetaData(Contentlet contentlet)
+                                contentMetaFile.getAbsolutePath(), e.getMessage()), e);
+            }
+        }
+
+        //If the metadata file does not exist we need to parse and get the metadata for the file
+        if (!contentMetaFile.exists()) {
+
+            if (binaryField != null) {
+
+                //Parse the metadata from this file
+                metaDataMap =
+                        this.getForcedMetaDataMap(binaryField, metadataFields,
+                                Config.getIntProperty("binaryfield.metadata.content.maxlength", 1024));
+
+                this.writeCompressJsonMetadataFile (contentMetaFile,
+                        UtilMethods.isSet(metaDataMap)?metaDataMap:Collections.emptyMap());
+            }
+        } else {
+
+            metaDataMap = this.readCompressedJsonMetadataFile (contentMetaFile);
+        }
+
+        return metaDataMap;
+    }
+
+    private InputStream createInputStream(final Path path) throws IOException {
+
+        // compressor config
+        final String compressor = Config
+                .getStringProperty("CONTENT_METADATA_COMPRESSOR", "none");
+
+        switch (compressor) {
+
+            case "gzip":
+                return new GZIPInputStream(Files.newInputStream(path));
+            case "bzip2":
+                return new BZip2CompressorInputStream(Files.newInputStream(path));
+            default:
+                return Files.newInputStream(path);
+        }
+    }
+
+    private OutputStream createOutputStream(final Path path) throws IOException {
+
+        // compressor config
+        final String compressor = Config
+                .getStringProperty("CONTENT_METADATA_COMPRESSOR", "none");
+
+        switch (compressor) {
+
+            case "gzip":
+                return new GZIPOutputStream(Files.newOutputStream(path));
+            case "bzip2":
+                return new BZip2CompressorOutputStream(Files.newOutputStream(path));
+            default:
+                return Files.newOutputStream(path);
+        }
+    }
+
+    private Map<String, Object> readCompressedJsonMetadataFile(final File contentMetaFile) {
+
+        InputStream inputStream = null;
+        Map<String, Object> objectMap = Collections.emptyMap();
+        try {
+
+            inputStream = this.createInputStream(contentMetaFile.toPath());
+            objectMap   = new GsonBuilder().disableHtmlEscaping().
+                    create().fromJson(new InputStreamReader(inputStream), Map.class);
+
+            Logger.info(this, "Metadata read from: " + contentMetaFile);
+        } catch (IOException e) {
+
+            Logger.error(this, e.getMessage(), e);
+        } finally {
+
+            IOUtils.closeQuietly(inputStream);
+        }
+
+        return objectMap;
+    }
+
+    private void writeCompressJsonMetadataFile(final File contentMetaFile, final Map<?, Object> objectMap) {
+
+        OutputStream out = null;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Ensure that resources like this OutputStream object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4960665)